fhir-server-test/src/test/java/com/ibm/fhir/server/test/BundleTest.java

```diff
@@ -2209,8 +2209,22 @@ public void testBatchConditionalDeletes() throws Exception {
         assertResponseBundle(responseBundle, BundleType.BATCH_RESPONSE, 4);
         assertGoodGetResponse(responseBundle.getEntry().get(0), Status.OK.getStatusCode(), HTTPReturnPreference.MINIMAL);
         assertGoodGetResponse(responseBundle.getEntry().get(1), Status.NO_CONTENT.getStatusCode(), HTTPReturnPreference.MINIMAL);
-        assertBadResponse(responseBundle.getEntry().get(2), Status.PRECONDITION_FAILED.getStatusCode(),
-                "returned multiple matches");
+
+        // A search that results in multiple matches:
+        // (1) if matches > FHIRConstants.FHIR_CONDITIONAL_DELETE_MAX_NUMBER_DEFAULT, then result in a 400 status code.
```

Comment:
should it be a 412 to be more consistent with the fhir spec?

Reply:
> Multiple matches: A server may choose to delete all the matching resources, or it may choose to return a 412 Precondition Failed error indicating the client's criteria were not selective enough. A server indicates whether it can delete multiple resources in its Capability Statement (.rest.resource.conditionalDelete)

Reply:
I choose not to use 412 just because I thought 412 means we don't support multiple matches at all. but we does support. anyway, I'm ok to change to use 412 instead.

Reply:
done